cylc/rose/rose.py

```diff
@@ -143,9 +145,27 @@ def get_rose_vars_from_config_node(config, config_node, environ):
 
         # Add standard ROSE_VARIABLES
         config_node[section].set(['ROSE_SITE'], rose_site)
-        config_node[section].set(['ROSE_VERSION'], ROSE_VERSION)
         config_node[section].set(['ROSE_ORIG_HOST'], rose_orig_host)
 
+        # Remove CYLC_VERSION to prevent conflict with Cylc 8.
+        if 'CYLC_VERSION' in config_node[section]:
+            LOG.warning(
+                f'CYLC_VERSION={config_node[section]["CYLC_VERSION"].value} '
+                'from rose-suite.conf not added to '
+                'processed conf to prevent conflict with Cylc 8 settings.'
+            )
+            del config_node[section]['CYLC_VERSION']
+        # If ROSE_VERSION is not the version we are using warn and replace.
+        if 'ROSE_VERSION' in config_node[section]:
+            if config_node[section]['ROSE_VERSION'] != ROSE_VERSION:
+                LOG.warning(
+                    'ROSE_VERSION='
+                    f'{config_node[section]["ROSE_VERSION"].value} '
+                    f'replaced with {ROSE_VERSION}: That is the version you '
+                    'are using.'
+                )
```

Comment:
And for this one, I would go for something along the lines of:
```suggestion
            LOG.warning(
                'ROSE_VERSION='
                f'{config_node[section]["ROSE_VERSION"].value} '
                'in rose-suite.conf will be ignored. You are '
                f'using version {ROSE_VERSION}.'
            )
```
I would get rid of the `if`; otherwise, if someone puts `ROSE_VERSION=2`, they won't get a warning and won't know not to do this